### PR TITLE
Feature TMC#101 prototype support for observers

### PR DESCRIPTION
**Pull request by Lurkerpas**

### CODE REVIEW

File: sdl2promela/sdl/model.py [hunk 1]
```diff
@@ -221,8 +222,12 @@ class Parameter:
     target_variable: VariableReference
     """Target variable reference."""
 
+    declared_type: object
```
Comment (rphh) [inline sdl2promela/sdl/model.py:225]: I started the use of `object` for types from opengeode, but after some time I think that the correct one is `Any` https://docs.python.org/3/library/typing.html#typing.Any

File: sdl2promela/sdl/model.py [hunk 2]
```diff
@@ -602,6 +617,42 @@ def __init__(self):
...
+    senderName: str
+    """Name of the Sender."""
+
+    recipientName: str
+    """Name of the Recipient."""
+
+    def __init__(self):
+        self.observerSignalName = ""
+        self.originalSignalName = ""
+        self.kind = ObservedSignalKind.INPUT
```
Comment (rphh) [inline sdl2promela/sdl/model.py:651]: Why this one value is default? Maybe `None` would be better here?

File: sdl2promela/sdl/model.py [hunk 3]
```diff
@@ -726,14 +777,18 @@ def convert(source: ogAST.TaskForLoop):
...
 def convert(source: ogAST.TaskAssign):
-    if len(source.elems) != 1:
-        raise ValueError(
-            "Assignment with an unsupported number of elements: " + len(source.elems)
-        )
-    task = AssignmentTask()
-    task.assignment = convert(source.elems[0])
-    task.type = source.elems[0].left.exprType
-    return task
+    if len(source.elems) == 1:
```
Comment (rphh) [inline sdl2promela/sdl/model.py:780]: Is it possible to eliminate this special case and use only for loop?

File: sdl2promela/sdl/model.py [hunk 4]
```diff
@@ -1102,8 +1157,15 @@ class Model:
     The value is a tuple where first element is type and the second
     is initial variable value.
     """
+    implicit_variables: Dict[str, Tuple[object, object]]
```
Comment (rphh) [inline sdl2promela/sdl/model.py:1160]: `typing.Any` instead of object

File: sdl2promela/translator.py [hunk 5]
```diff
@@ -1424,6 +1454,14 @@ def __generate_transition_function(context: Context) -> promelamodel.Inline:
     return builder.build()
 
 
+def __generate_implicit_variable_definition(
+    context: Context, variable_name: str, variable_type: Tuple[object, object]
```
Comment (rphh) [inline sdl2promela/translator.py:1458]: `Any`

File: tests/integration/input_observer/Makefile [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+# glue code generated by Kazoo), they are wrapped inside "host" Promela
+# files
+
+input_observer:
+	echo "Test input_observer"
+	${CREATE_EMPTY_TEST_DIR}
+	cp ${RES_DIR}/system_structure.pr ${TEST_DIR}
+	cp ${RES_DIR}/InputObserver.pr ${TEST_DIR}
+	cp ${RES_DIR}/dataview-uniq.asn ${TEST_DIR}
+	cd  ${TEST_DIR} && opengeode --toAda system_structure.pr InputObserver.pr
```
Comment (rphh) [inline tests/integration/input_observer/Makefile:44]: What is the purpose of --toAda here?

Comment (Lurkerpas) [reply tests/integration/input_observer/Makefile:44]: To generate dataview

**Review by rphh — CHANGES_REQUESTED** [review]
Few improvements

File: sdl2promela/sdl/model.py (hunk 4 above)
Comment (Lurkerpas) [reply sdl2promela/sdl/model.py:1160]: Done

File: sdl2promela/translator.py (hunk 5 above)
Comment (Lurkerpas) [reply sdl2promela/translator.py:1458]: Done

File: sdl2promela/sdl/model.py (hunk 3 above)
Comment (rphh) [reply sdl2promela/sdl/model.py:780]: ok

**Pull request merged**